api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/BaseDao.java

```diff
@@ -443,6 +453,49 @@ protected boolean containsAlias(Iterator<CriteriaImpl.Subcriteria> subcriteriaIt
 		return Optional.empty();
 	}
 	
+	/**
+	 * A handler for a {@link HasParam}
+	 *
+	 * @param propertyName the name of the property in the query to use
+	 * @param hasParam the {@link HasParam} to handle
+	 * @return a {@link Criterion} to be added to the query for the indicate has param
+	 */
+	protected Optional<Criterion> handleHas(String propertyName, HasParam hasParam) {
+		if (hasParam == null) {
+			return Optional.empty();
+		}
+		ArrayList<IQueryParameterType> orValues = new ArrayList<>();
+		String targetResourceType = null;
+		String paramReference = null;
+		String parameterName = null;
+		
+		String paramName = null;
+		targetResourceType = hasParam.getTargetResourceType();
+		
+		if (targetResourceType.getClass().equals(Observation.class)) {
+			paramReference = hasParam.getReferenceFieldName();
+			parameterName = hasParam.getParameterName();
+			paramName = parameterName.replaceAll("\\..*", "");
+			hasParam = new HasParam(targetResourceType, paramReference, parameterName, hasParam.getParameterValue());
+			orValues.add(hasParam);
+			orValues.add(
+			    (IQueryParameterType) QualifiedParamList.singleton(null, hasParam.getValueAsQueryToken(fhirContext)));
+		}
+		String qualifier = paramName.substring(4);
+		hasParam.setValueAsQueryToken(fhirContext, FhirConstants.HAS_PROPERTY, qualifier,
+		    hasParam.getValueAsQueryToken(fhirContext));
+		orValues.add(hasParam);
+		return Optional.of(eq(propertyName, orValues));
```

Comment:
Does this actually work? It seems to me like this is never going to match.

Reply:
true ,,made some changes